jobs/payment-jobs/tasks/unpaid_statement_notify_task.py

```diff
@@ -0,0 +1,131 @@
+
+            for account_id in auth_account_ids:
+                try:
+                    # Get the most recent monthly statement
+                    statement = cls.find_most_recent_statement(account_id, StatementFrequency.MONTHLY.value)
+                    summary = cls.get_statement_owing(account_id, statement.id)
+                    payment_account: PaymentAccountModel = PaymentAccountModel.find_by_id(statement.payment_account_id)
+                    recipients = StatementRecipientsModel.\
+                        find_all_recipients_for_payment_id(statement.payment_account_id)
+
+                    if len(recipients) < 1:
+                        current_app.logger.info(f'No recipients found for statement: '
+                                                f'{statement.payment_account_id}.Skipping sending')
+                        continue
+
+                    to_emails = ','.join([str(recipient.email) for recipient in recipients])
+
+                    # Send payment notification if payment account is using EFT and there is an amount owing
+                    if payment_account.payment_method == PaymentMethod.EFT.value and summary['total_due'] > 0:
+                        publish_payment_notification(pay_account=payment_account,
+                                                     statement=statement,
+                                                     is_due=is_due,
+                                                     due_date=last_day.date(),
+                                                     emails=to_emails)
+
+                except Exception as e:  # NOQA # pylint: disable=broad-except
+                    capture_message(
+                        f'Error on unpaid statement notification auth_account_id={account_id}, '
+                        f'ERROR : {str(e)}', level='error')
+                    current_app.logger.error(e)
+                    continue
+
+    @classmethod
+    def find_most_recent_statement(cls, auth_account_id: str, statement_frequency: str) -> StatementModel:
+        """Find all payment and invoices specific to a statement."""
+        query = db.session.query(StatementModel) \
+            .join(PaymentAccountModel, PaymentAccountModel.auth_account_id == auth_account_id) \
+            .filter(StatementModel.frequency == statement_frequency) \
+            .order_by(StatementModel.to_date.desc())
+
+        return query.first()
+
+    @classmethod
+    def get_statement_owing(cls, auth_account_id: str, statement_id: str):
+        """Get statement owing amount by account id."""
+        result = db.session.query(func.sum(InvoiceModel.total - InvoiceModel.paid).label('total_due')) \
+            .join(PaymentAccountModel) \
+            .join(StatementInvoicesModel) \
+            .filter(PaymentAccountModel.auth_account_id == auth_account_id) \
+            .filter(InvoiceModel.invoice_status_code.in_((InvoiceStatus.SETTLEMENT_SCHEDULED.value,
```

Comment:
Since we are sending out reminders 7 days before and on the due date, I don't believe they are technically OVERDUE yet, so I included the additional statuses based on the unpaid invoice notify job.